cljfmt/src/cljfmt/core.cljc

```diff
@@ -568,10 +568,16 @@
 #?(:clj
    (defn- as-zloc->alias-mapping [as-zloc]
      (let [alias             (some-> as-zloc z/right z/sexpr)
-           current-namespace (some-> as-zloc z/leftmost z/sexpr)
+           current-namespace (some-> as-zloc
+                                     z/leftmost
+                                     (z/find #(n/symbol-node?
+                                               (z/node (skip-meta %))))
+                                     z/sexpr)
            grandparent-node  (some-> as-zloc z/up z/up)
            parent-namespace  (when-not (ns-require-form? grandparent-node)
-                               (first (z/child-sexprs grandparent-node)))]
+                               (when (or (z/vector? grandparent-node)
```

Comment:
if we have something like

```clj
^{:clj-kondo/ignore [:discouraged-namespace} [my.namespace :as n]
```

then `grandparent-node` in this case is a `:meta` node so by adding the vector or string check we prevent that from accidentally returning an alias map like 

```clj
{"n" "{:clj-kondo/ignore [:discouraged-namespace]}.my.namespace"}
```